components/kf-monitoring/README.md

```diff
@@ -0,0 +1,22 @@
+# kf-monitoring
+
+kf-monitoring is common monitoring util library for Kubeflow components written in go.
+The goal is to provide interface for both metrics generators (kubeflow components) and metrics consumters (alerts / UI). 
+When upstream components record metrics through kf-monitoring, new metrics will be picked up by downstream consumers without changing downstream configs.
+
+**Benefit of using kf-monitoring**
+
+kf-monitoring defines:
+- Fixed metrics names and types
+  - Metrics names is consumed by downstream components who is configured by metrics names
+  - Two major downstream consumers:
+    - [alert rules](https://prometheus.io/docs/prometheus/latest/configuration/alerting_rules/) that keep scanning metrics
+    - [monitoring dashboard](https://prometheus.io/docs/visualization/grafana/) that display metrics groups
+- structured label group
+  - Each component use labels to identify the metrics they are recording.
+  - Current label group: `component, kind, namespace, action, severity`
+  
+**Integration instruction**
+
+WIP
```

Comment:
For easy integration, is it possible to integrate this with Istio for services with sensible defaults? (This may overlap with Istio dashboard) e.g.

component -> Istio virtual service name
kind -> "endpointStatus"
action -> Get/Post/... + endpointPath
severity -> 4XX(major) / 5XX(critical)

I think it is hard for some component to change its source code to use this metrics server. For example, Metadata uses MLMD gRPC docker image out of box. Even if some component can, it would be valuable if the external behavior is captured by default and developers only need to capture the internal status.